tests/nodes/tests_DBpediaMode.cpp

```diff
@@ -0,0 +1,60 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include <doctest/doctest.h>
+
+#include <rdf4cpp/rdf.hpp>
+
+using namespace rdf4cpp::rdf;
+
+TEST_CASE("IRI") {
+    IRIFactory f{};
+    static constexpr std::string_view iri = "http//:example:foo/bar";
+
+    datatypes::registry::dbpedia_mode = false;
+    CHECK(!f.create_and_validate(iri).has_value());
+
+    datatypes::registry::dbpedia_mode = true;
+    CHECK(f.create_and_validate(iri).has_value());
+    CHECK(f.create_and_validate(iri).value().identifier() == iri);
+}
+
+TEST_CASE("integers") {
+    Literal u{};
+    datatypes::registry::dbpedia_mode = false;
+    CHECK_THROWS_AS(u = Literal::make_typed<datatypes::xsd::NonNegativeInteger>("-5"), std::runtime_error);
+    CHECK_THROWS_AS(u = Literal::make_typed<datatypes::xsd::PositiveInteger>("0"), std::runtime_error);
+    CHECK(u == Literal{});
+
+    datatypes::registry::dbpedia_mode = true;
+    CHECK(Literal::make_typed<datatypes::xsd::NonNegativeInteger>("-5") == Literal::make_typed<datatypes::xsd::NonNegativeInteger>("0"));
+    CHECK(Literal::make_typed<datatypes::xsd::PositiveInteger>("0") == Literal::make_typed<datatypes::xsd::PositiveInteger>("1"));
+}
+
+TEST_CASE("date/time") {
+    Literal u{};
+    datatypes::registry::dbpedia_mode = false;
+    CHECK_THROWS_AS(u = Literal::make_typed<datatypes::xsd::GDay>("---40"), std::runtime_error);
+    CHECK_THROWS_AS(u = Literal::make_typed<datatypes::xsd::GMonth>("--40"), std::runtime_error);
+    CHECK_THROWS_AS(u = Literal::make_typed<datatypes::xsd::GMonthDay>("--40-40"), std::runtime_error);
+    CHECK_THROWS_AS(u = Literal::make_typed<datatypes::xsd::GYearMonth>("1742-0"), std::runtime_error);
+    CHECK_THROWS_AS(u = Literal::make_typed<datatypes::xsd::Date>("1742-0-0"), std::runtime_error);
+    CHECK_THROWS_AS(u = Literal::make_typed<datatypes::xsd::Date>("1742-2-40"), std::runtime_error);
+    CHECK_THROWS_AS(u = Literal::make_typed<datatypes::xsd::Time>("10:90:00"), std::runtime_error);
+    CHECK_THROWS_AS(u = Literal::make_typed<datatypes::xsd::Time>("26:90:60"), std::runtime_error);
+    CHECK_THROWS_AS(u = Literal::make_typed<datatypes::xsd::DateTime>("1742-2-40T26:90:60"), std::runtime_error);
+    CHECK_THROWS_AS(u = Literal::make_typed<datatypes::xsd::DateTimeStamp>("1742-2-40T26:90:60Z"), std::runtime_error);
+    CHECK(u == Literal{});
+
+    datatypes::registry::dbpedia_mode = true;
+    CHECK(Literal::make_typed<datatypes::xsd::GDay>("---40") == Literal::make_typed<datatypes::xsd::GDay>("---31"));
```

Comment:
Same here, we don't really know what they meant by that. They probably just had a buggy datetime implementation that just added days together without checking if it makes sense that way.

But since this is gDay we don't really have a way to fix what they meant => discard